phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -985,7 +976,17 @@ public TableRef refreshDerivedTableNode(DerivedTableNode derivedTableNode) throw
               return this.resolveTable(null, tableAlias);
         }
 
-        private static class ColumnFamilyRef {
+        protected void rememberTable(TableRef tableRef, String alias) {
+            String name = tableRef.getTable().getName().getString();
+            if (alias != null) {
+                tableMap.put(alias, tableRef);
+            } else {
+                tableMap.put(name, tableRef);
+            }
+            tables.add(tableRef);
+        }
+
+        protected static class ColumnFamilyRef {
```

Comment:
This can be private?

Reply:
No, we use it in child ProjectedTableColumnResolver.

Reply:
Actually, It doesn't do anything. 
I have removed both ColumnFamilyRef and resolveColumnFamily, which is never calles.